encoding/encoding.go

```diff
@@ -69,6 +69,13 @@ type Serializer interface {
 	MethodType() MethodType
 }
 
+// ProtoErrorSerializer deserializes errors.
+// Error details are very specific to proto.
+type ProtoErrorSerializer interface {
```

Comment:
Nit: should we name this interface `ProtoErrorDetailsDecoder` and its method as `Decode`?

Reply:
I went with `ProtoErrorDeserializer` to stay consistent with the other interface `Serializer`